internal/storage/store.go

```diff
@@ -16,6 +16,9 @@ type KVStore interface {
 	io.Closer
 	// Put stores the association between the given key and value
 	Put(key []byte, value []byte) error
+	// PutTTL stores the association between the given key and value
+	// and sets the expireTS of the key to the provided epoch in seconds
+	PutTTL(key []byte, value []byte, expireTS int64) error
```

Comment:
Consider converting `expireTS` to `uint64` since it really is that. Avoids few type casts in the impl. code too.

Reply:
Converting to uint64 needs way more typecast as `time.Now()` returns int64 which then in all places needs to be typecased.